cgen error for missing _option_ok without `-no-skip-unused`

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.11 198ecd0, press to see full `v doctor` output</summary>

|V full version      |V 0.4.11 f03d80080016c880d40601ca2fcb94795155a62a.198ecd0
|:-------------------|:-------------------
|OS                  |linux, Ubuntu 20.04.6 LTS
|Processor           |4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz
|Memory              |1.25GB/15.05GB
|                    |
|V executable        |/home/delian/v/v
|V last modified time|2025-08-02 08:09:44
|                    |
|V home dir          |OK, value: /home/delian/v
|VMODULES            |OK, value: /home/delian/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/delian/v
|                    |
|Git version         |git version 2.50.0
|V git status        |weekly.2025.31-34-g198ecd09
|.git/config present |true
|                    |
|cc version          |cc (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
|gcc version         |gcc (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
|clang version       |N/A
|tcc version         |tcc version 0.9.28rc 2025-02-13 HEAD@f8bd136d (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 696c1d84
|emcc version        |N/A
|glibc version       |ldd (Ubuntu GLIBC 2.31-0ubuntu9.18) 2.31

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  j.v && j`
```v
import json
type Any = string | f32 | bool | ?int
x := json.decode([]Any, '["hi", -9.8e7, true, null]')!
dump(x)

```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/j.01K1MX998R32EPEM255DQHTJ5D.tmp.c:1813: warning: implicit declaration of function '_option_ok'
cc: /home/delian/v/thirdparty/tcc/lib/libgc.a: error: 'GC_noop1_ptr' defined twice
cc: tcc: error: undefined symbol '_option_ok'
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
```

**What did you expect to see?**

a compiled program


> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/json/tests/json_decode_sumtype_option_test.v
+++ b/vlib/json/tests/json_decode_sumtype_option_test.v
@@ -1,0 +1,8 @@
+import json
+
+type Any = string | f32 | bool | ?int
+
+fn test_main() {
+	x := json.decode([]Any, '["hi", -9.8e7, true, null]')!
+	assert dump(x) == [Any('hi'), Any(f32(-9.8e+7)), Any(true), Any('')]
+}

--- a/vlib/v/markused/walker.v
+++ b/vlib/v/markused/walker.v
@@ -1128,6 +1128,9 @@ pub fn (mut w Walker) mark_by_sym(isym ast.TypeSymbol) {
 					w.features.used_maps++
 					continue
 				}
+				if typ.has_flag(.option) {
+					w.used_option++
+				}
 				w.mark_by_type(typ)
 			}
 		}

